### PR TITLE
Implement get_device_info for F5359 used by KPN Box 12

### PR DESCRIPTION
**Pull request by rhpijnacker**
Getting the value for Path `Device/DeviceInfo` triggers the `UnknownPathException` on a F5359 device.
The required information is available via individual requests, though.

### CODE REVIEW

**Comment by rhpijnacker** [comment]
An obvious improvement is to create one combined request for all of these values.
As a first start and to trigger the discussion I am sending the pull request like this.

**Comment by iMicknl** [comment]
> An obvious improvement is to create one combined request for all of these values. As a first start and to trigger the discussion I am sending the pull request like this.

Thanks! I wonder why this doesn't work while pulling Device Info. 

Wouldn't it make sense to catch this exception in the HA integration and execute the different commands?
(thus move the logic from this PR to ha-sagemcom)

**Comment by rhpijnacker** [comment]
> Wouldn't it make sense to catch this exception in the HA integration and execute the different commands?
(thus move the logic from this PR to ha-sagemcom)

I do not think so. Conceptually this still returns the device info, so seems to fit well where I put it.
I'm not sure about some of the fields, though. Do you have an example of what it looks like for another device?

**Comment by rhpijnacker** [comment]
> I wonder why this doesn't work while pulling Device Info.

I would not know either. We can try to raise a bug report at KPN, but I'm not optimistic that will lead anywhere.

**Comment by iMicknl** [comment]
Thanks @rhpijnacker, I will try to look at this early next week. Would be good if it could be merged and released :).

**Comment by rhpijnacker** [comment]
@iMicknl To clean up things, I also implemented a single-call with multiple actions interface. I just did not yet create a PR for that. You may want to have a look at that too: https://github.com/rhpijnacker/python-sagemcom-api/tree/feature/f5359-device-info-2. I could merge this if you think this is an improvement.

**Comment by rhpijnacker** [comment]
@iMicknl 

> Would be good if it could be merged and released :).

That would be nice indeed!
With the X-mas holidays coming up, I'd have time to do some rework (if needed).
I'm eagerly anticipating your comments.

**Comment by iMicknl** [comment]
@rhpijnacker sorry I didn't review this one yet, since CI is failing. Do you know how to run pre-commit locally? This will help you with the styling etc.

Otherwise I can help out and merge it afterwards.

**Comment by rhpijnacker** [comment]
@iMicknl No, I do not know how to run the pre-commit locally (yet).
I was waiting on feedback first on whether or not this is the right direction in the first place.

**Review by iMicknl — COMMENTED** [review]
PR is fine by me! However, the login/logout part is in this PR as well.

Could you reply on the logout pr? If we can agree on that one, I will merge both and release :).

**Comment by rhpijnacker** [comment]
Yeah, I think I committed to the wrong branch at some point, which explains why the logout part is in here too.

Before releasing this, you may want to take a look at https://github.com/rhpijnacker/python-sagemcom-api/tree/feature/f5359-device-info-2 too. This combines the call to get the device info in one rest call, instead of doing multiple.

**Comment by iMicknl** [comment]
> Before releasing this, you may want to take a look at [rhpijnacker@55adf7f](https://github.com/rhpijnacker/python-sagemcom-api/commit/55adf7f2bceefdcd269fa7ff62df740122c12412) too. This combines the call to get the device info in one rest call, instead of doing multiple.

This would be a good improvement for a follow up PR. As you see, I am a bit late with replies lately since I am quite busy with bringing another integration to Home Assistant core. (and other things outside HA :D).

I will merge this one now and see if we need something more, otherwise I can release it.

File: sagemcom_api/client.py
Comment (iMicknl) [inline sagemcom_api/client.py:346]: In a follow up PR, it would be good to try catch this as well, since this could also throw an `UnknownPathException`.

**Review by iMicknl — APPROVED** [review]

File: sagemcom_api/client.py
Comment (rhpijnacker) [reply sagemcom_api/client.py:346]: What do you propose to do if this also throws an exception?
I do not see any way to recover from that, so why not just let the exception propagate?

Comment (iMicknl) [reply sagemcom_api/client.py:346]: You are right, we need to catch this in the Home Assistant integration :-)

**Pull request merged**
